### ISSUE TITLE
Engine update check may run at every startup

### ISSUE THREAD

**Issue by ssoloff** [triplea-game/triplea#3914]
### Engine version

1.9.0.0.11651

### Description

A fresh install of TripleA will set `TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY` to `true`.  When this setting is `true`, regardless of when the engine update check was last run (as determined by the `TRIPLEA_LAST_CHECK_FOR_ENGINE_UPDATE` setting), the engine update check will always run.

`TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY` is never automatically reset to `false` (e.g. after the engine update check), so **the default behavior of a fresh install of TripleA will be to run the engine update check at every startup**.  Whereas, the desired behavior, as coded, is to only check for engine updates once every two days (assuming it truly is not the first time TripleA is run).

AFAIK, the only way to change this behavior is to open the Engine Settings and set the **Game > Show First Time Prompts** setting to False.

The `TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY` setting was first added in 7b0fd0ec6.  Two things to note about the original implementation:

1. The setting name had the current engine version appended, thus creating a unique setting for each version of TripleA (i.e. every time a new version was released, an engine update check would be forced regardless of when it last occurred).
1. The setting value was reset to `false` immediately after a `true` value triggered an engine update check.

It appears this functionality was partially broken in each of 46b55e5e1 and 5a7e89270:

* 46b55e5e1 removed resetting the value of `TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY` to `false`.
* 5a7e89270 changed the name of `TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY` to a constant that no longer depends on engine version.  Without the unique suffix, the meaning of the setting essentially becomes "is this the first time we're running _any_ version of TripleA?"

It would be fairly easy to add back in resetting the value to `false` during the engine update check.  However, I'm not sure if restoring the version-specific nature of this setting is required.

**Comment by RoiEXLab** [comment]
@ssoloff There's also another problem to consider:
https://github.com/triplea-game/triplea/blob/6ca38e86355d9bebca6de3866d574c8c7a63b1c1/latest_version.properties#L1
Is the file being checked for the new version, as you may notice, there are only 4 parts in the version number, because the build number we use is really just a permanent workaround for a consistent version scheme.
So in order to fix the update checker we'd need to either change the way the file is parsed (I think it's parsed into a Version object which only supports 4 dot separated numbers, I might be wrong though), or finally rework the versioning system to make the build number an actual part of it.

**Comment by ssoloff** [comment]
@RoiEXLab Good point.  However, my understanding is that we explicitly do not prompt users to upgrade simply based on build number.  I don't have any reference to point to.  I simply remember some discussions that happened early last year (2017) when we were talking about bumping the fourth digit in the version.

Perhaps @ron-murhammer or @DanVanAtta can clarify what the desired behavior is here.

**Comment by ron-murhammer** [comment]
Yeah, we don't want to prompt users to upgrade to every pre-release build we put out. We really only want to prompt when we put out a new stable (essentially when we update the website download).

**Comment by RoiEXLab** [comment]
@ron-murhammer
The problem stays though: Even when we release a new stable, we couldn't even change the file to point to it, so users will never know when there's a new stable out.
We could do the same thing as the website and query github directly, but I believe this might exceed the maximum limit of requests for the github api easily, whem the client is restarted too many times.

**Comment by DanVanAtta** [comment]
This is resolved, the engine check was rewritten.
